### ISSUE TITLE
Trajectory Analysis

### ISSUE THREAD

**Issue by obasekigit** [prody/ProDy#1695]
Good day all, i have noticed for quite some time now that ProDy has not been able to calculate dynamic data from my trajectory like rmsd, radius of gyration etc

**Comment by obasekigit** [comment]
![Screenshot from 2023-05-11 08-42-33](https://github.com/prody/ProDy/assets/84424732/e4d0aac1-9839-471e-bec5-c786c443d4e0)

**Comment by karolamik13** [comment]
Good day, 
you might consider using parseDCD() instead:

from prody import *
PDBfile = 'x.pdb'
DCDfile = 'x.dcd'

structure = parsePDB(PDBfile)
ensemble = parseDCD(DCDfile)
ensemble.setAtoms(structure)
ensemble.setCoords(structure)

structure = structure.select('segment PROA and resid 162')
ensemble.setAtoms(structure)
ensemble.superpose()    
rmsd = ensemble.getRMSDs()
rmsf = ensemble.getRMSFs()

Best wishes, Karolina

**Comment by obasekigit** [comment]
The problem with using parseDCD is that it cannot calculate other parameters like distance and gyration

**Comment by jamesmkrieger** [comment]
I think the problem could be that you're only providing one object to calcRMSD. Have you tried `calcRMSD(frame, ntd_a)`?

**Comment by jamesmkrieger** [comment]
You'd probably also have to use setAtoms first

**Comment by karolamik13** [comment]
I think we know why it is not working. Your script should work but because of some improvements in the code which was done quite recently, it stopped working correctly. I think that all the time you are analyzing the same frame which is why you have zeros (RMSD for the same structure). The code will work when we will fix the bug in Trajectory().

**Comment by jamesmkrieger** [comment]
> The problem with using parseDCD is that it cannot calculate other parameters like distance and gyration

calcDistance and calcGyradius are general functions that you can use on both Trajectory (also from DCDFile) and Ensemble (from parseDCD). 

calcGyradius is just as easy with both and gives the same answers:
```
In [8]: gyr = zeros(traj.numFrames())

In [9]: for i, frame in enumerate(traj):
   ...:     gyr[i] = calcGyradius(frame)
   ...: 

In [10]: ens = parseDCD('prody/tests/datafiles/dcd2k39_truncated.dcd')
@> DCD file contains 3 coordinate sets for 167 atoms.
@> DCD file was parsed in 0.00 seconds.
@> 0.01 MB parsed at input rate 9.81 MB/s.
@> 3 coordinate sets parsed at input rate 4936 frame/s.

In [11]: gyr
Out[11]: array([8.9517537 , 8.98492115, 8.95398523])

In [12]: gyr = zeros(traj.numFrames())

In [13]: gyr
Out[13]: array([0., 0., 0.])

In [14]: for i, frame in enumerate(ens):
    ...:     gyr[i] = calcGyradius(frame)
    ...: 

In [15]: gyr
Out[15]: array([8.9517537 , 8.98492115, 8.95398523])
```

For distance (which actually isn't in the tutorial at http://prody.csb.pitt.edu/tutorials/trajectory_analysis/trajectory.html even though it says it should be), it's obvious easier to use Trajectory because of the linked atoms getting updated coordinates. Trajectory does have the disadvantage that once you've looped through frames then you need to reset it, but that's ok.
```
In [16]: ntd_a = structure.select('resid 1 and name CA')

In [17]: ntd_b = structure.select('resid 5 and name CA')

In [18]: dist = zeros(traj.numFrames())

In [19]: traj.reset()

In [20]: for i, frame in enumerate(traj):
    ...:     dist[i] = calcDistance(ntd_a, ntd_b)
    ...: 

In [21]: dist
Out[21]: array([13.17431164, 13.53827763, 13.01749706])

In [22]: traj.reset()

In [23]: ntd_a = structure.select('resid 1 and name CA')

In [24]: ntd_b = structure.select('resid 8 and name CA')

In [25]: for i, frame in enumerate(traj):
    ...:     dist[i] = calcDistance(ntd_a, ntd_b)
    ...: 

In [26]: dist
Out[26]: array([23.02369499, 23.5089035 , 22.90761566])
```

With Ensemble, the way to do it would be to use setAtoms many times extract things into variables separately and then you need to use setAtoms again at the end:
```
In [28]: ntd_a = structure.select('resid 1 and name CA')

In [29]: ntd_b = structure.select('resid 5 and name CA')

In [30]: for i, frame in enumerate(ens):
    ...:     ens.setAtoms(ntd_a)
    ...:     ntd_a_pos = frame.getCoords()
    ...: 
    ...:     ens.setAtoms(ntd_b)
    ...:     ntd_b_pos = frame.getCoords()
    ...: 
    ...:     dist[i] = calcDistance(ntd_a_pos, ntd_b_pos)
    ...: 

In [31]: dist
Out[31]: array([13.17431164, 13.53827763, 13.01749706])

In [32]: ens.setAtoms(structure)
```

So, in summary, they have their advantages and disadvantages but should both work, I think

**Comment by jamesmkrieger** [comment]
Please feel free to re-open this issue if something isn't clear or you think there's still a problem. For now, I am closing it
